code/modules/projectiles/guns/syringe_gun.dm

```diff
@@ -95,3 +94,280 @@
 	force = 2 //Also very weak because it's smaller
 	suppressed = 1 //Softer fire sound
 	can_unsuppress = 0 //Permanently silenced
+
+// Not quite a syringe gun, but also not completely unlike one either.
+/obj/item/gun/syringe/rapidsyringe
+	name = "rapid syringe gun"
+	desc = "A syndicate rapid syringe gun based on an archived NanoTrasen prototype. Capable of storing and filling syringes from an internal reservoir.\nIt has a large flap on the side that you can dump a box or bag of syringes into, and a port for filling it with liquid."
+	icon_state = "rapidsyringegun"
+	max_syringes = 14  // full two boxes worth
+	/// Index of possible transfer amounts to select
+	var/transfer_amount_selection = 1
+	/// Maximum size of the internal reservoir
+	var/reservoir_volume = 300
+	/// Possible options for alt-clicking the
+	var/list/possible_transfer_amounts = list(5, 10, 15)
+	/// Amount of reagents to transfer out at once if no syringe is loaded
+	var/reservoir_transfer_amount = 15
+	/// Whether or not we've alerted the user that the reservoir is empty.
+	var/alarmed = FALSE
+	container_type = 0  // We'll handle it ourselves tyvm
+
+/obj/item/gun/syringe/rapidsyringe/Initialize(mapload)
+	. = ..()
+	create_reagents(reservoir_volume)
+
+/obj/item/gun/syringe/rapidsyringe/Destroy()
+	for(var/syringe in syringes)
+		qdel(syringe)
+
+	syringes = null
```

Comment:
instead of doing this, you can do `QDEL_LIST(syringes)` and it will delete the whole list, it won't null it out so it will still be `syringes = list()` but that shouldn't matter.